### ISSUE TITLE
Turn on the fan and huge delay of command execution

### ISSUE THREAD

**Issue by Lyr3x** [Condorello/Domoticz-Xiaomi-Fan-ZA4#1]
While testing the `miio_fan_server.sh` i found out that this script does nothing. No response at all. However running the python-server directly the plugin is working partly.

I can turn the fan off but it wont turn on anymore. Do you have the same issue?

**Comment by Condorello** [comment]
Hi, honestly not, it works fine here. Which domoticz version are you running? I'm on latest stable.
And my fan have a reserved dhcp address on my router, so it's like a static ip.

**Comment by Lyr3x** [comment]
> Which domoticz version are you running?

latest beta

> And my fan have a reserved dhcp address on my router, so it's like a static ip.

Same for me.

I just realized that it took over 3 minutes that the fan reacts to commands. It will turn on fine but most commands are have a huge delay. Is that something you experience too?

The commands via the mi home app are executed immediately. Also when i am using the `miiocli` everything responds quick

**Comment by Lyr3x** [comment]
Digging more inot it i found out the following. 

```
server: got socket msg: ['status']
server: fan result {'power_state': 'on', 'battery': None, 'fan_level': 100, 'oscillate_state': True, 'angle_state': 60, 'fan_mode_state': 'direct', 'cmd': 'status'}
server: send reply ('127.0.0.1', 46022)
```

this part seems blocking somehow. After the send_replay i could send one command immedately without any delay. Then i need to wayt for the next stus message processing.

When the Server sends the `status` request everything else is blocked and will be processed afterwards like here:
```
server: send reply ('127.0.0.1', 46022)
server: got socket msg: ['status']
server: got socket msg: ['set_fan_level_direct', 100]
server: got socket msg: ['set_fan_level_direct', 100]
server: got socket msg: ['set_fan_level_direct', 100]
server: got socket msg: ['set_fan_level_direct', 100]
server: got socket msg: ['set_fan_level_direct', 100]
server: fan result {'power_state': 'on', 'battery': None, 'fan_level': 100, 'oscillate_state': True, 'angle_state': 60, 'fan_mode_state': 'natural', 'cmd': 'status'}
server: send reply ('127.0.0.1', 46022)
server: fan result {'code': ['ok'], 'cmd': 'set_fan_level_direct'}
server: send reply ('127.0.0.1', 46022)
server: fan result {'code': ['ok'], 'cmd': 'set_fan_level_direct'}
server: send reply ('127.0.0.1', 46022)
server: fan result {'code': ['ok'], 'cmd': 'set_fan_level_direct'}
server: send reply ('127.0.0.1', 46022)
server: fan result {'code': ['ok'], 'cmd': 'set_fan_level_direct'}
server: send reply ('127.0.0.1', 46022)
server: fan result {'code': ['ok'], 'cmd': 'set_fan_level_direct'}
server: send reply ('127.0.0.1', 46022)
server: got socket msg: ['status']
```

when sending one command the whole server is blocked by `server: send reply ('127.0.0.1', 47900)`

**Comment by Condorello** [comment]
In the hardware tab of domoticz try to increase update interval to 30 and
After 60 and see of less call to 'status' give better response


Il sab 1 ago 2020, 16:19 Kai Bepperling <notifications@github.com> ha
scritto:

**Comment by Lyr3x** [comment]
Yep already increased it to 60 and it repsonses quicker if not hitting the status window :)
I just realized the the status request tooks very long and it is not a particular problem with the plugin itself... Could you build in a toggle to disable the whole status update code? I personally would prefer a smooth control with inapproriate state at the moment. 

Even with a higher update interval sometimes it keeps blocking for minutes and does not respond to commands :/
An asynchronous implementation of the status update would work. I just tested it "manually". The fan will execute any command wil the cli requests the latest state.
